be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/MessageEncoder.scala

```diff
@@ -0,0 +1,19 @@
+package ch.epfl.pop.pubsub.graph
+
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import ch.epfl.pop.json.JsonMessageParser
+import ch.epfl.pop.model.network.JsonRpcMessage
+
+import scala.util.Try
+
+object MessageEncoder {
+
+  val serializer: Flow[JsonRpcMessage, Nothing, NotUsed] = Flow[JsonRpcMessage].map { message =>
+    val serializedMessage: Try[String] = Try(JsonMessageParser.serializeMessage(message))
```

Comment:
Wait a second, there are 2 messages that need to be sent out at this point.
One is the broadcast (if we received a publish), the other is the response.
I don't think we defined the two (or more flows) following a publish, leading to sending:
- a broadcast to other clients
- a response to the publishing client
- a future as-of-yet-undefined message to other servers

Reply:
If I understand correctly, this could be the job of the "answerer" in the diagram above. Depending on the situation, it might send 0 to N messages. I do not know yet how to contact others (broadcast), nor how to serialize the answers to be sent (since the encoder is tehcnically located before in the pipeline)

Reply:
I think the broadcast might be an additional "sink" from the LaoHandler (and others in that partition/merge group), following a separate route, as not all messages will be broadcast on the same channel (e.g. Lao/create creates its own channel, and the broadcast happens on that channel)